src/sentry/api/endpoints/project_rules_configuration.py

```diff
@@ -25,6 +26,11 @@ def get(self, request, project):
             if hasattr(node, "form_fields"):
                 context["formFields"] = node.form_fields
 
+            # It is possible for a project to have no services. In that scenario we do not want the front end to render
+            # the action as the action does not have options.
+            if isinstance(node, NotifyEventServiceAction) and len(node.get_services()) == 0:
```

Comment:
What does the FE look like if we return an empty `action_list` and `condition_list`?

Reply:
I'm not quite sure what this line is doing. If you had no services, wouldn't the alert rule have no actions anyway?

Reply:
@wedamija  I have to get back to you.
@scefali Thanks for asking. So this is what we're trying to achieve (No services -> NotifyEventServiceAction does not exist for the user). Currently, if you have no services, the action is rendered without any services as options

Reply:
It's now possible for us to have no services here, since mail is always active but counts as a separate action now. I've updated this pr after merging the rest of jeff's work, will merge this soon.